java/opendcs/src/main/java/ilex/util/FileLogger.java

```diff
@@ -87,25 +88,33 @@ public FileLogger( String procName, String filename )
 	* @param maxLength Maximum length of log file
 	* @throws FileNotFoundException if can't open file
 	*/
-	public FileLogger( String procName, String filename, int maxLength )
-		throws FileNotFoundException
+	public FileLogger(String procName, String filename, int maxLength) throws FileNotFoundException
 	{
 		super(procName);
 		this.filename = EnvExpander.expand(filename);
 		this.maxLength = maxLength;
+		isTerminal = ("/dev/stdout".equals(filename) || "CON:".equalsIgnoreCase(filename));
 		openNewLog();
 		writerThread = new Thread(() ->
 			{
-				while(closeOperations.get() == false)
+				while (closeOperations.get() == false)
 				{
 					try
 					{
 						String msg = queue.poll(1, TimeUnit.SECONDS);
 						PrintStream ps = output.get();
 						if (msg != null && ps != null && !ps.checkError())
 						{
+							if (!isTerminal) // avoid any kind of buffer overflow errors since this would just keep accumulating.
+							{
+								bytesWritten.getAndAdd(msg.length());
+							}
 							ps.println(msg);
 						}
+						if (!isTerminal && (bytesWritten.get() >= this.maxLength))
```

Comment:
Any chance we can get configurable log file lengths for OpenDCS programs, or is this whole file logging function being re-written elsewhere?

Reply:
@agilmore2  - are you thinking about a feature where decodes.properties, or command line argument allows setting the max log file size?